docs/resources/domain_record.md

```diff
@@ -0,0 +1,176 @@
+  priority = 20
+}
+```
+
+### With dynamic records
+
+```hcl
+resource "scaleway_domain_record" "geo_ip" {
+  dns_zone = "domain.tld"
+  name     = "images"
+  type     = "A"
+  data     = "1.2.3.4"
+  ttl      = 3600
+  geo_ip {
+    matches {
+      continents = ["EU"]
+      countries  = ["FR"]
+      data       = "1.2.3.5"
+    }
+    matches {
+      continents = ["NA"]
+      data       = "4.3.2.1"
+    }
+  }
+}
+
+resource "scaleway_domain_record" "http_service" {
+  dns_zone = "domain.tld"
+  name     = "app"
+  type     = "A"
+  data     = "1.2.3.4"
+  ttl      = 3600
+  http_service {
+    ips          = ["1.2.3.5", "1.2.3.6"]
+    must_contain = "up"
+    url          = "http://mywebsite.com/health"
+    user_agent   = "scw_service_up"
+    strategy     = "hashed"
+  }
+}
+
+resource "scaleway_domain_record" "view" {
+  dns_zone = "domain.tld"
+  name     = "db"
+  type     = "A"
+  data     = "1.2.3.4"
+  ttl      = 3600
+  view {
+    subnet = "100.0.0.0/16"
+    data   = "1.2.3.5"
+  }
+  view {
+    subnet = "100.1.0.0/16"
+    data   = "1.2.3.6"
+  }
+}
+
+resource "scaleway_domain_record" "weighted" {
+  dns_zone = "domain.tld"
+  name     = "web"
+  type     = "A"
+  data     = "1.2.3.4"
+  ttl      = 3600
+  weighted {
+    ip     = "1.2.3.5"
+    weight = 1
+  }
+  weighted {
+    ip     = "1.2.3.6"
+    weight = 2
+  }
+}
+```
+
+## Arguments Reference
+
+The following arguments are supported:
+
+- `dns_zone` - (Required) The DNS Zone of the domain. If the DNS zone doesn't exist, it will be automatically created.
+
+- `keep_empty_zone` - (Optional, default: `false`) When destroying a resource, if only NS records remain and this is set to `false`, the zone will be deleted. Please note, each zone not deleted will [cost you money](https://www.scaleway.com/en/dns/)
+
+- `name` - (Required) The name of the record (can be an empty string for a root record).
+
+- `type` - (Required) The type of the record (`A`, `AAAA`, `MX`, `CNAME`, `ALIAS`, `NS`, `PTR`, `SRV`, `TXT`, `TLSA`, or `CAA`).
+
+- `data` - (Required) The content of the record (an IPv4 for an `A`, a string for a `TXT`...).
+
+- `ttl` - (Optional, default: `3600`) Time To Tive of the record.
+  
+- `priority` - (Optional, default: `0`) The priority of the record (mostly used with an `MX` record)
+
+**Dynamic records:**
+
+- `geo_ip` - (Optional) The Geo IP feature provides DNS resolution, based on the user’s geographical location. You can define a default IP that resolves if no Geo IP rule matches, and specify IPs for each geographical zone. [Documentation and usage example](https://www.scaleway.com/en/docs/scaleway-dns/#-Geo-IP-Records)
+    - `matches` - (Required) The list of matches. *(Can be more than 1)*
+        - `countries` - (Optional) List of countries (eg: `FR` for France, `US` for the United States, `GB` for Great Britain...)
+        - `continents` - (Optional) List of continents (eg: `EU` for Europe, `NA` for North America, `AS` for Asia...)
```

Comment:
What is the name of the norm used? Could you add a link to the exact values used?